draft-ietf-ppm-dap.md

```diff
@@ -1514,217 +1607,232 @@ to the next phase of the aggregation protocol.
-`leader_prep_share` in the next round of continuation). For the latter case, the
-helper sets `prep_state` to `next_prep_state`.
+where `report_id` is the report ID associated with `state` and `outbound`.
 
-The Leader now advances its aggregation job to the next round (round 1 if this
-is the first continuation after initialization) and then instructs the Helper to
-advance the aggregation job to the round the Leader has just reached by sending
-the new `prep_msg` message to the Helper in a POST request to the aggregation
-job URI used during initialization (see {{leader-init}}). The body of the
-request is an `AggregationJobContinueReq`:
+Next, the Leader sends a POST request to the aggregation job URI used during
+initialization (see {{leader-init}}) with media type
+"application/dap-aggregation-job-continue-req" and body structured as:
 
 ~~~
 struct {
-  u16 round;
+  u16 step;
   PrepareStep prepare_steps<1..2^32-1>;
 } AggregationJobContinueReq;
 ~~~
 
-The `round` field is the round of VDAF preparation that the Leader just reached
-and wants the Helper to advance to.
+The `step` field is the step of DAP aggregation that the Leader just reached
+and wants the Helper to advance to. The `prepare_steps` field is the sequence
+of prep steps constructed in the previous step. The `PrepareStep`s MUST be in
+the same order as the previous aggregate request.
 
-The `prepare_steps` field MUST be a sequence of `PrepareStep`s in the
-`continued` state containing the corresponding `inbound` prepare message.
+The Helper's response will be an `AggregationJobResp` message (see
+{{aggregation-helper-init}}), which the Leader validates according to the
+criteria in {{aggregation-job-validation}}. If the message is invalid, the
+Leader MUST abort the aggregation job.
 
-The media type is set to "application/dap-aggregation-job-continue-req".
+Otherwise, the Leader proceeds as follows with each report:
 
-The Helper's response will be an `AggregationJobResp` message (see
-{{aggregation-helper-init}}), which the Leader validates according to
-the criteria in {{aggregation-job-validation}}. If the message is valid, the
-Leader moves to the next round of continuation with the enclosed prepare steps.
-Otherwise, the Leader should abandon the aggregation job entirely.
+1. If the inbound prep step type is "continue", then the Leader computes
+
+   ~~~
+   (state, outbound) = VDAF.ping_pong_req(agg_param, state, inbound)
+   ~~~
+
+   where `inbound` is the payload of the prep step. If `state == Rejected()`,
+   then the Leader rejects the report and removes it from the candidate set. If
+   `state == Continued(prep_state)`, then another continuation step is required;
+   the Leader stores `state` and `outbound` for the next continuation step. If
+   `state == Finished(out_share)`, then preparation is complete and the Leader
+   stores the output share for use in the collection flow ({{collect-flow}}).
```

Comment:
This is also incorrect I'm pretty sure: if `outbound != None` but `state == Finished(out_share)` then we need to be sure to send the last outbound message.